apps/rush-lib/src/utilities/LastInstallFlag.ts

```diff
@@ -0,0 +1,58 @@
+import * as path from 'path';
+import * as fsx from 'fs-extra';
+import * as _ from 'lodash';
+
+export const LAST_INSTALL_FLAG_FILE_NAME: string = 'last-install.flag';
+
+/**
+ * A helper class for managing last-install flags, which are persistent and
+ * indicate that something installed in the folder was sucessfully completed.
+ * It also compares state, so that if something like the Node JS version has changed,
+ * it can invalidate the last install.
+ * @internal
+ */
+export class LastInstallFlag {
+  private _flagPath: string;
+
+  /**
+   * Creates a new LastInstall flag
+   * @param folderPath - the folder that this flag is managing
+   * @param _state - optional, the state that should be managed or compared
+  */
+  constructor(folderPath: string, private _state: Object = {}) {
```

Comment:
>private [](start = 34, length = 7)

Never use this language feature.  It hides the data fields, which are arguably the most important aspect of a class.  (The thesis of object-oriented programming is to focus on data instead of operations; if the nouns are clearly defined then the interpretation of the verbs will be obvious.)